content/blog/2022-03-31-training-and-saving-models-on-a-dedicated-runner.md

```diff
@@ -0,0 +1,278 @@
+---
+title: Training and saving models with CML on a dedicated AWS EC2 runner
+date: 2022-03-31
+description:
+  In this guide we will show how you can use CML to automatically retrain a
+  model and save its outputs to your Github repository using a provisioned AWS
+  EC2 runner.
+descriptionLong: |
+  We can use CML to automatically retrain models whenever data, model code,
+  or parameters change. In this guide we show how to create a pipeline that
+  provisions an AWS EC2 instance to retrain a model and save the output on
+  a regular basis. This way we can prevent drift by ensuring that our model
+  always uses the latest input data.
+picture: 2015-05-01/post-image.jpeg
+# pictureComment: Some _Comment_ (supports _basic_ [Markdown](link))
+author: rob_dewit
+# commentsUrl: https://discuss.dvc.org/t/february-22-community-gems/1078
+tags:
+  - CML
+  - Git
+  - Pipelines
+  - Dedicated runners
+  - AWS
+---
+
+When you first develop a machine learning model, you will probably do so on your
+local machine. You can easily change algorithms, parameters, and input data
+right in your text editor, notebook, or terminal. Imagine you have a
+long-running model for which you want to detect possible
+[drift](https://en.wikipedia.org/wiki/Concept_drift), however. In that case it
+would be beneficial to automatically retrain your model on a regular basis.
+
+In this guide, we will show how you can use
+[CML (Continuous Machine Learning)](https://cml.dev/) to do just that. CML is an
+open-source library for implementing continuous integration and delivery (CI/CD)
+in machine learning projects. This way we can define a pipeline to train a model
+and keep track of various versions. Although we could do so directly in our
+CI/CD pipeline (e.g. GitHub Workflows), the runners used for this generally
+don’t have a lot of processing power. Therefore it makes more sense to provision
+a dedicated runner that is tailored to our computing needs.
+
+At the end of this guide we will have set up a CML workflow that does the
+following on a daily basis:
+
+1. Provision an Amazon Web Services (AWS) EC2 instance
+1. Train the model
+1. Save the model and its metrics to a GitHub repository
+1. Terminate the AWS EC2 instance
+
+In a follow-up post we will expand upon this by using [DVC](https://dvc.org/) to
+designate a remote storage for our resulting models. But let's focus on CML
+first!
+
+All files needed for this guide can be found in
+[this repository](https://github.com/iterative/example_model_export_cml).
+
+<admon type="info">
+This guide can be followed on its own, but also as an extension to this <a href="https://cml.dev/doc/self-hosted-runners">example in the docs</a>.
+</admon>
+
+<admon type="tip">
+We wil be using GitHub for our CI/CD and AWS for our computing resources. With slight modifications, you should be able to use Gitlab and Google Cloud or Microsoft Azure respectively.
```

Comment:
should be able -> "would be able to" or "can"
Should makes it seem questionable.